src/ruby/puppet-server-lib/puppet/server/config.rb

```diff
@@ -25,6 +25,15 @@ def self.initialize_puppet_server(puppet_server_config)
       Puppet::Util::Profiler.add_profiler(@profiler)
     end
 
+    if puppet_server_config["shared_terminus_state"]
+      begin
+        require 'puppet/util/puppetdb/shared_terminus_state'
+        Puppet::Util::Puppetdb::SharedTerminusState.shared_terminus_state = puppet_server_config["shared_terminus_state"]
```

Comment:
I like @cprice404's idea about making this a map of state maps.  If we keep it PuppetDB specific for now, I'd be tempted to just call it `:puppetdb-shared-terminus-state` instead elsewhere in the Clojure code.  But either way, fine with it as it is for now.